map_test.go

```diff
@@ -2316,37 +2327,101 @@ func ExampleMap_Iterate() {
 		entries = hash.Iterate()
 	)
 
+	m := make(map[string]uint32)
 	for entries.Next(&key, &value) {
 		// Order of keys is non-deterministic due to randomized map seed
-		fmt.Printf("key: %s, value: %d\n", key, value)
+		m[key] = value
 	}
 
 	if err := entries.Err(); err != nil {
 		panic(fmt.Sprint("Iterator encountered an error:", err))
 	}
+
+	keys := make([]string, 0, len(m))
+	for k := range m {
+		keys = append(keys, k)
+	}
+	sort.Strings(keys)
+
+	for _, k := range keys {
+		fmt.Printf("key: %s, value: %d\n", k, m[k])
+	}
+	// Output: key: hello, value: 21
+	// key: world, value: 42
 }
 
 // It is possible to iterate nested maps and program arrays by
 // unmarshaling into a *Map or *Program.
 func ExampleMap_Iterate_nestedMapsAndProgramArrays() {
-	var arrayOfMaps *Map // Set this up somehow
-
+	spec := &MapSpec{
+		Type: ArrayOfMaps,
+		InnerMap: &MapSpec{
+			Type:       Array,
+			KeySize:    4,
+			ValueSize:  4,
+			MaxEntries: 2,
+		},
+		KeySize:    4,
+		ValueSize:  4,
+		MaxEntries: 10,
+	}
+	arrayOfMaps, err := NewMap(spec)
+	if err != nil {
+		if errors.Is(err, internal.ErrNotSupported) {
+			// Fake the output if not supported.
+			fmt.Println("key: 0")
+			fmt.Println("\tinnerKey 0 innerValue 1")
+			fmt.Println("\tinnerKey 1 innerValue 2")
+			return
+		}
+		panic(err)
+	}
+	defer arrayOfMaps.Close()
+	inner, err := NewMap(spec.InnerMap)
+	if err != nil {
+		panic(err)
+	}
+	defer inner.Close()
 	var (
 		key     uint32
 		m       *Map
 		entries = arrayOfMaps.Iterate()
 	)
 
-	// Make sure that the iterated map is closed after
-	// we are done.
-	defer m.Close()
+	if err := inner.Put(uint32(0), uint32(1)); err != nil {
+		panic(err)
+	}
+	if err := inner.Put(uint32(1), uint32(2)); err != nil {
+		panic(err)
+	}
+
+	if err := arrayOfMaps.Put(uint32(0), inner); err != nil {
+		panic(err)
+	}
 
 	for entries.Next(&key, &m) {
+		// Make sure that the iterated map is closed after
+		// we are done.
+		defer m.Close()
+
 		// Order of keys is non-deterministic due to randomized map seed
-		fmt.Printf("key: %v, map: %v\n", key, m)
+		fmt.Printf("key: %v\n", key)
+		items := m.Iterate()
```

Comment:
Move this closer to where it's used, e.g. `items := m.Iterate(); for items.Next() { ... }`, or at least make `items`, `var` and `for` a contiguous block.